veles/secrets/openai/validator_test.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openai_test
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/google/osv-scalibr/veles"
+	"github.com/google/osv-scalibr/veles/secrets/openai"
+)
+
+const (
+	projectValidatorTestKey = "sk-proj-12345678901234567890T3BlbkFJ" +
+		"12345678901234567890123456"
+)
+
+// mockOpenAIServer creates a mock OpenAI API server for testing project keys
+func mockOpenAIServer(t *testing.T, expectedKey string, statusCode int,
+	responseBody string) *httptest.Server {
```

Comment:
It doesn't look like the validator code actually looks at the responseBody. Do we need to set it in the tests?